html/janus.js

```diff
@@ -44,10 +44,109 @@ Janus.isExtensionEnabled = function() {
+			}
+			if(options.body !== undefined) {
+				fetchOptions.body = JSON.stringify(options.body);
+			}
+
+			var fetching = f(url, fetchOptions).then(function(response) {
+				if(response.ok) {
+					// minor optimisation: no need to parse response body if there is no success handler to do anything with it.
+					if(typeof(options.success) === typeof(Janus.noop)) {
+						return response.json().then(function(result) {
+							// success!
+							options.success(result);
+						}).catch(function(error) {
+							// JSON parsing error
+							if(typeof(options.error) === typeof(Janus.noop)) {
+								options.error(response.statusText, error);
+							}
+						});
+					}
+				}
+				else {
+					// API error (HTTP status code says so)
+					if(typeof(options.error) === typeof(Janus.noop)) {
+						options.error(response.statusText, response.status);
+					}
+				}
+			}).catch(function(error) { // network error
+				if(typeof(options.error) === typeof(Janus.noop)) {
+					options.error('<< network error >>', error);
+				}
+			});
+
+			/*
+			 * fetch() does not natively support timeouts.
+			 * Work around this by starting a timeout manually, and racing it agains the fetch() to see which thing resolves first.
+			 */
+			if(options.timeout !==  undefined) {
+				var timeoutError = 'request timeout';
+				var timeout = new p(function(resolve, reject) {
+					setTimeout(reject, options.timeout, timeoutError);
+				}).catch(function(error) { // timeout error
+					if(typeof(options.error) === typeof(Janus.noop)) {
+						options.error('<< network error >>', error);
+					}
+				});
+
+				fetching = p.race([fetching, timeout]);
+			}
+			return fetching;
+		}
+	}
+};
+
+Janus.useOldDependencies = function (jQueryObj, adapterObj) {
+	var jq = jQueryObj || $;
+	return {
+		isArray: function(arr) { return jq.isArray(arr); },
+		checkJanusExtendsion: function() { return jq('#janus-extension-installed').length > 0; },
+		webRTCAdapter: adapterObj || adapter,
+		httpAPICall: function(url, options) {
+			var payload = options.body !== undefined ? {
+				contentType: 'application/json',
+				data: JSON.stringify(options.body)
+			} : {};
+			var credentials = options.withCredentials !== undefined ? {xhrFields: {withCredentials: options.withCredentials}} : {};
+
+			return $.ajax($.extend(payload, credentials, {
+				url: url,
+				type: options.verb,
+				cache: false,
+				dataType: 'json',
+				async: options.async,
+				timeout: options.timeout,
+				success: function(result) {
+					if(typeof(options.success) === typeof(Janus.noop)) {
+						options.success(result);
+					}
+				},
+				error: function(xhr, status, err) {
+					if(typeof(options.error) === typeof(Janus.noop)) {
+						options.error(status, err);
+					}
+				}
+			}));
+		},
+	};
+};
+
 Janus.noop = function() {};
 
 // Initialization
-Janus.init = function(options) {
+Janus.init = function(options, deps) {
```

Comment:
Since you're checking for ```options.dependencies```, what's the purpose of providing the extra arg?

Reply:
Either one is sufficient by itself, I think this is a remnant from various attempts I've played with while putting the PR together.

Reply:
Then I'd recommend ripping out the extra arg, and leaving the ```dependencies``` key in ```options```